darts/timeseries.py

```diff
@@ -3005,6 +3037,598 @@ def map(
+                )
+
+        # validate tansformation dictionary format
+        if window_transformations is None:
+            raise_log(
+                ValueError(
+                    "window_transformations argument should be provided and "
+                    "must be a non-empty dictionary or a non-empty list of dictionaries."
+                ),
+                logger,
+            )
+
+        if window_transformations is not None:
+
+            raise_if_not(
+                (
+                    isinstance(window_transformations, dict)
+                    and len(window_transformations) > 0
+                )
+                or (
+                    isinstance(window_transformations, list)
+                    and len(window_transformations) > 0
+                ),
+                "`window_transformations` must be a non-empty dictionary or a non-empty list of dictionaries. ",
+            )
+
+            if isinstance(window_transformations, dict):
+                window_transformations = [
+                    window_transformations
+                ]  # if only one dictionary, make it a list
+
+            for idx, transformation in enumerate(window_transformations):
+                raise_if_not(
+                    isinstance(transformation, dict),
+                    "`window_transformations` must contain dictionaries.",
+                )
+
+                raise_if_not(
+                    "function" in transformation
+                    and transformation["function"] is not None
+                    and (
+                        isinstance(transformation["function"], Callable)
+                        or (transformation["function"] in _builtin_transforms)
+                    ),
+                    f"`window_transformation` dictionary at index {idx} must contain the 'function' key and be "
+                    f"callable or one of {list(_builtin_transforms.keys())}.",
+                )
+
+                if isinstance(transformation["function"], str):
+                    raise_if_not(
+                        # test that mandatory arguments for builtins are provided
+                        # (tests that intersection between two sets is equal to the set of mandatory arguments)
+                        transformation["function"] in _builtin_transforms
+                        and set(
+                            itertools.chain(
+                                *_builtin_transforms[transformation["function"]][2]
+                            )
+                        ).intersection(set(transformation.keys()))
+                        == set(
+                            itertools.chain(
+                                *_builtin_transforms[transformation["function"]][2]
+                            )
+                        ),
+                        f"`window_transformation` dictionary at index {idx} must contain at least the following keys "
```

Comment:
the error message that would be raised here for missing arguments , without the above code, would come from pandas. I have no problem with that. I thought it might be clearer/easier to control the set of mandatory arguments on Darts side and notify the user, but I see that this is creating more redundancy with pandas than true usefulness. I can remove those checks and this indeed removes the need to generate the _builtin_transforms dictionary, which main use was to check the mandatory arguments!